src/monitor.rs

```diff
@@ -942,6 +942,7 @@ where
 
         let pec = PEC15::calc(&result[0..6]);
         if pec[0] != result[6] || pec[1] != result[7] {
+            self.cs.set_high().map_err(Error::CSPinError)?;
```

Comment:
From my point of view, it's the correct & expected behavior for setting the CS pine state here.
As the only upper method calling `read()` is within `read_daisy_chain()`, which is managing the CS pin state (except the case you just found).